### DISCUSSION TITLE
CompositeController take control of existing resource as Child

### DISCUSSION THREAD

**Discussion by zgcalebp** [metacontroller/metacontroller#849] [Q&A]
Hello!

I am working on adding a CompositeController for a custom resource, and I am hoping to have it behave such that if a child resource returned from the sync hook already exists on the kubernetes cluster, that the existing resource will be appropriately updated to now be owned and maintained by my CompositeController.

Right now, the metacontroller logs the following error:
```
Failed to create child, child object already exists.
```

Is there a configuration I can change on either the CompositeController or the Metacontroller to enable this behavior? If not, are there any recommendations for how to handle this use case in a mostly seamless way?

Thanks!

**Comment by grzesuav** [comment]
hi @zgcalebp , actually not sure, can you ping me later on #meacontroller slack channel ? Would need to check it. Does the existing child have `ownerRef` ?

**Reply by andrewj8080** [reply]
Hello too!

Sorry to revive a months old post. I've just ran into this same question. Did you guys find a solution? 

Thank you!

**Edit:** My guess is I can just add the parent controller-uid label to each resource and that's probably it.

**Reply by grzesuav** [reply]
hmm I belive it is adoptin existing object....can you provide some short example ? Ping me also on #metacontrroller slack channel
